Support Kotlin data classes for describing type-safe Qute templates

### Description

### It would be great if we could use a Kotlin data class to describe a type-safe Qute template.

Originally posted as a [question on StackOverflow](https://stackoverflow.com/questions/77816595/type-safe-qute-template-with-kotlin-data-classes), here's the full story.

I'm feeding paginated data into a Qute template. The data is of type `Paginated<out T>`, where `Paginated` adds information about total page count, current page, etc.

The [documentation](https://quarkus.io/guides/qute-reference#typesafe_templates) says you can let your controller return

> a Java record that implements io.quarkus.qute.TemplateInstance

And I was hoping that I could do the same thing with a Kotlin data class. I prefer that over invoking an `@JvmStatic external fun` method since the latter is a lot harder to write unit tests for.

I'd like to have my controller methods return instances of that data class:

```kotlin
@GET @Path("/list") @Produces(MediaType.TEXT_HTML)
fun list(@QueryParam("pageIndex") pageIndex: Int = 0): TemplateInstance {
  // work your magic
  return ExampleListTemplate(
    Paginated(
      currentPage = 0,
      pageCount = 1,
      items = listOf("example")
    )
  )
}

@CheckedTemplate(basePath = "example")
data class ExampleListTemplate(val data: Paginated<String>): TemplateInstance
```

### Implementation ideas


Current behaviour is that if you request such a page, you'll get a stacktrace:

```
Caused by: java.lang.UnsupportedOperationException
    at io.quarkus.qute.TemplateInstance.getTemplate(TemplateInstance.java:142)
    at io.quarkus.resteasy.qute.runtime.TemplateResponseFilter.filter(TemplateResponseFilter.java:45)
```

This is because the data class doesn't override the `getAttribute` (default) method from the `TemplateInstance` interface.

This is where I get a little bit lost. If I followed the Java record approach, my record wouldn't override that default method either, but it would work nevertheless. I guess that by studying how it works for Java records, it might be possible to find a solution direction for Kotlin data classes, but I didn't walk that far (yet).

--- a/docs/src/main/asciidoc/qute-reference.adoc
+++ b/docs/src/main/asciidoc/qute-reference.adoc
@@ -2055,6 +2055,21 @@ public class HelloResource {
 <1> Declares a type-safe template with the Java record. The template is located at `/src/main/resources/templates/HelloResource/Hello.html`.
 <2> Instantiate the record and use it as an ordinary `TemplateInstance`.
 
+[TIP]
+====
+**Kotlin users** can use a data class annotated with `@JvmRecord` to achieve the same result:
+
+[source,kotlin]
+----
+@CheckedTemplate(basePath = "HelloResource")
+@JvmRecord
+data class Hello(val name: String) : TemplateInstance
+----
+
+The `@JvmRecord` annotation instructs the Kotlin compiler to generate a Java record in the bytecode, which Quarkus then recognizes as a type-safe template.
+Note that `@JvmRecord` requires Kotlin 1.5+ and JVM target 16+.
+====
+
 
 ==== Customized Template Path
 

